src/components/tasks/TaskSearch/TaskSearch.tsx

```diff
@@ -16,84 +23,316 @@ const initialSuggestionCoordinates: SuggestionCoordinates = {
+        toggleInputFocus();
+        onClickSearchButton(convertSearchOptionsToQuery(updatedOptions));
+    };
+
+    const onSuggestionSelected = (
+        idx = activeFilterSuggestionDropdownIndex
+    ) => {
+        if (!filterSuggestions?.[idx]) return;
+
+        if (onEditSelectedFilterIndex === false) {
+            const optionDetails = filterSuggestions[idx];
+            if (optionDetails) {
+                setSelectedFilters([...selectedFilters, optionDetails]);
+                setTypedInput('');
+                onClickSearchButton(
+                    convertSearchOptionsToQuery([
+                        ...selectedFilters,
+                        optionDetails,
+                    ])
+                );
+            }
+        } else {
+            const newOptions = selectedFilters;
+            newOptions[onEditSelectedFilterIndex] = filterSuggestions[idx];
+            setSelectedFilters(newOptions);
+            setOnEditSelectedFilterIndex(false);
+            onClickSearchButton(convertSearchOptionsToQuery(newOptions));
+        }
+        setActiveFilterSuggestionDropdownIndex(-1);
+        toggleInputFocus(true);
+    };
+
+    const handleClickOutside = (event: React.MouseEvent<HTMLInputElement>) => {
+        const target = event.target as HTMLElement;
+        if (
+            target &&
+            target.className.includes('pill-input-wrapper') &&
+            onEditSelectedFilterIndex !== false
+        ) {
+            setOnEditSelectedFilterIndex(false);
+        } else if (target && target.className.includes('pill-input-wrapper')) {
+            setOnEditSelectedFilterValue('');
+            setOnRemoveSelectedFilterIndex(-1);
+            toggleInputFocus();
+        }
+    };
+    useEffect(() => {
+        if (onEditSelectedFilterIndex === false) {
+            toggleInputFocus();
+            setOnRemoveSelectedFilterIndex(-1);
+            setOnEditSelectedFilterValue('');
+        }
+    }, [onEditSelectedFilterIndex]);
+
+    useEffect(() => {
+        filterSuggestions.length === 0 &&
+            setSuggestionCoordinates(initialSuggestionCoordinates);
+    }, [filterSuggestions.length]);
+
+    useEffect(() => {
+        window.addEventListener('resize', onResizeHandler);
+    }, []);
     return (
         <div className={styles['task-search-container']}>
             <div id="filter-container" className={styles['filter-container']}>
                 <div className={styles['filter-button']} onClick={handleModal}>
-                    <p>Filter</p>
-                    <LuChevronDown
-                        className={` ${
-                            modalOpen
-                                ? styles['filter-chevron-open']
-                                : styles['filter-chevron']
-                        }`}
-                    />
-                    {modalOpen && (
-                        <FilterModal
-                            dev={false}
+                    Filter
```

Comment:
Can such a long JSX be broken down into multiple functions? That would considerably improve the readability.